ui/app/routes/workspace/projects/project/app/build.ts

```diff
@@ -28,11 +28,12 @@ export default class BuildDetail extends Route {
 
   async model(params: BuildModelParams) {
     // Setup the build request
+    let { builds } = this.modelFor('workspace.projects.project.app');
+    let { id: build_id } = builds.find((obj) => obj.sequence == params.sequence);
```

Comment:
[sand] Let’s use threequals here just to avoid any javascript typecasting funny business.

```suggestion
    let { id: build_id } = builds.find((obj) => obj.sequence === params.sequence);
```

Reply:
Used the `==` because the threequals was not passing. Looked into the reason why that happened a bit more, and for some reason the parameter's `sequence` value gets converted to a string at some point in the process of coming down to the build router (even though I have the interface saying it should be a number), any ideas why that would happen?

Reply:
Ahhh, very interesting! I actually think this is a great example of where the implicit type-coercion of `==` can lead to trouble. Without `===`, we wouldn’t have realized that we were dealing with mixed types, and might have wound up with some weird coercion bugs.

As to why the parameter comes through as a string rather than a number — I think that’s the only permitted behavior of Ember’s router. I _think_ it’s because that parameter originates in the URL, so therefore it starts off as a string and should really stay a string unless we do some explicit conversion.

So, TL;DR I think having an explicit type conversion here would be the ideal. Something like:

```ts
obj.sequence === Number(params.sequence)
```